### PR TITLE
[tests-only][full-ci] add test coverage to open open-document file with /app/open-with-web endpoint

### PR DESCRIPTION
**Pull request by anon-pradip**
## Description

Added test to open open-document file with /app/open-with-web endpoint with and without app_name parameter.

## Related Issue

- Related to: https://github.com/owncloud/ocis/issues/9682

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -43,6 +43,14 @@ public static function isUUIDv4(string $id): bool {
...
 	}
 
+	/**
+	 *
+	 * @return string
+	 */
+	public static function getFileUUIDv4Regex(): string {
+		return self::getUUIDv4Regex() . '%24' . self::getUUIDv4Regex() . '%21' . self::getUUIDv4Regex();
+	}
+
```
Comment (PrajwolAmatya) [inline tests/acceptance/TestHelpers/GraphHelper.php:53]:
```suggestion
```

I think this won't be necessary we can assert uuidv4 regex with encoded value in the json schema something as:
```
fileId=%uuidv4_pattern%%24%uuidv4_pattern%%21%uuidv4_pattern%
```

File: tests/acceptance/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -2317,6 +2317,14 @@ public function substituteInLineCodes(
 					"getUUIDv4Regex"
 				],
 				"parameter" => []
+			],
+			[
+				"code" => "%file_uuidv4_pattern%",
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/FeatureContext.php:2322]:
```suggestion
				"code" => "%uuidv4_pattern%",
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 3]
```diff
@@ -171,3 +171,30 @@ Feature: collaboration (wopi)
       | app-endpoint                                     |
       | /app/open?file_id=<<FILEID>>&app_name=FakeOffice |
       | /app/open?file_id=<<FILEID>>                     |
+
+
+  Scenario Outline: open file with .odt extension in web
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:176]:
```suggestion
  @issue-9495
  Scenario Outline: open file with .odt extension (open-with-web)
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 4]
```diff
@@ -171,3 +171,30 @@ Feature: collaboration (wopi)
...
+      """
+      {
+        "type": "object",
+        "required": [
+          "uri"
+        ],
+        "properties": {
+          "uri": {
+            "type": "string",
+             "pattern": "^.*?contextRouteName=files-spaces-personal&fileId=%file_uuidv4_pattern%%24%file_uuidv4_pattern%%21%file_uuidv4_pattern%$"
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:191]: what's the actual URL here?

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 5]
```diff
@@ -171,3 +171,30 @@ Feature: collaboration (wopi)
...
+          }
+        }
+      }
+      """
+
+    Examples:
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:197]:
```suggestion
      """
    Examples:
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiCollaboration/wopi.feature (hunk 4 above)
Comment (anon-pradip) [reply tests/acceptance/features/apiCollaboration/wopi.feature:191]: If included `app_name` in endpoint then it is:
`https://host.docker.internal:9200/external?app=FakeOffice\u0026contextRouteName=files-spaces-personal\u0026fileId=c3ce162e-6dd4-4abc-b402-d50d385a0573%24d08915f8-9abb-4996-aa37-dec314d8ff05%219720f782-289e-49a9-aa98-718cdf9dd3fd`

else:

`https://host.docker.internal:9200/external?contextRouteName=files-spaces-personal\u0026fileId=c3ce162e-6dd4-4abc-b402-d50d385a0573%247bd222b5-1d3d-4cb4-8de6-c84bf1f0aa30%210994538e-bece-4234-9acc-547b1417613b`

Comment (saw-jan) [reply tests/acceptance/features/apiCollaboration/wopi.feature:191]: then pattern can be
`"pattern": "%base_url%/external?contextRouteName=...`

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 6]
```diff
@@ -171,3 +171,29 @@ Feature: collaboration (wopi)
...
+      """
+      {
+        "type": "object",
+        "required": [
+          "uri"
+        ],
+        "properties": {
+          "uri": {
+            "type": "string",
+             "pattern": "^.*?contextRouteName=files-spaces-personal&fileId=%uuidv4_pattern%%24%uuidv4_pattern%%21%uuidv4_pattern%$"
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:191]: please, check the full URL and all query params here.

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 7]
```diff
@@ -251,3 +251,29 @@ Feature: collaboration (wopi)
...
+            "type": "string",
+             "pattern": "%base_url%/external\\?(?:app=FakeOffice&)?contextRouteName=files-spaces-personal&fileId=%uuidv4_pattern%%24%uuidv4_pattern%%21%uuidv4_pattern%$"
+          }
+        }
+      }
+      """
+    Examples:
+      | app-endpoint                                              |
+      | /app/open-with-web?file_id=<<FILEID>>&app_name=FakeOffice |
+      | /app/open-with-web?file_id=<<FILEID>>                     |
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:279]:
```suggestion
      | app-endpoint                                              | url-query       |
      | /app/open-with-web?file_id=<<FILEID>>&app_name=FakeOffice | app=FakeOffice& |
      | /app/open-with-web?file_id=<<FILEID>>                     |                 |
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 8]
```diff
@@ -251,3 +251,29 @@ Feature: collaboration (wopi)
...
+      """
+      {
+        "type": "object",
+        "required": [
+          "uri"
+        ],
+        "properties": {
+          "uri": {
+            "type": "string",
+             "pattern": "%base_url%/external\\?(?:app=FakeOffice&)?contextRouteName=files-spaces-personal&fileId=%uuidv4_pattern%%24%uuidv4_pattern%%21%uuidv4_pattern%$"
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:271]: if that works
```suggestion
             "pattern": "%base_url%/external\\?<url-query>contextRouteName=files-spaces-personal&fileId=%uuidv4_pattern%%24%uuidv4_pattern%%21%uuidv4_pattern%$"
```

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Review by PrajwolAmatya — APPROVED** [review]

**Pull request merged**